Sources/AWSSDKSwiftCore/AWSClient.swift

```diff
@@ -547,12 +604,194 @@ extension AWSClient {
+
+        var baggage: BaggageContext = .init()
+
+        internal init(logger: Logger, baggage: BaggageContext = .init()) {
+            self._logger = logger
+            self.baggage = baggage
+        }
+    }
+
+    private static let loggingDisabled = Logger(label: "AWS-do-not-log", factory: { _ in SwiftLogNoOpLogHandler() })
+
+    public static func emptyContext(logger: Logging.Logger? = nil, baggage: BaggageContext = .init()) -> AWSClient.Context {
+        AWSClient.DefaultContext(logger: logger ?? Self.loggingDisabled, baggage: baggage)
+    }
+}
+
+extension AWSClient.Context {
+    // TODO: discuss how to update context baggage
+    public func with(baggage: BaggageContext) -> AWSClient.Context {
+        var copy = self
+        copy.baggage = baggage
+        return copy
+    }
+}
+
+// MARK: RequestMetadata
+
+// extension Logger {
+//    func attachingRequestId(_ id: Int, operation: String, service: String) -> Logger {
+//        var logger = self
+//        logger[metadataKey: "aws-service"] = .string(service)
+//        logger[metadataKey: "aws-operation"] = .string(operation)
+//        logger[metadataKey: "aws-request-id"] = "\(id)"
+//        return logger
+//    }
+// }
+
+private extension AWSClient {
+    struct RequestMetadata: CustomStringConvertible {
+        static let globalRequestID = NIOAtomic<Int>.makeAtomic(value: 0)
+
+//        static func create(
+//            operation: String,
+//            serviceConfig: AWSServiceConfig,
+//            requestId: Int = Self.globalRequestID.add(1)
+//        ) -> RequestMetadata {
+//            RequestMetadata(requestId: requestId, service: serviceConfig.service, operation: operation)
+//        }
+
+//        static func create(serviceConfig: AWSServiceConfig) -> (String) -> RequestMetadata {
+//            { operation in
+//                Self.create(operation: operation, serviceConfig: serviceConfig)
+//            }
+//        }
+
+        var requestId: Int
+        var service: String
+        var operation: String
+
+        var description: String {
+            "aws-request-id=\(requestId),aws-service=\(service),aws-operation=\(operation)"
+        }
+
+        init(operation: String, serviceConfig: AWSServiceConfig, requestId: Int = Self.globalRequestID.add(1)) {
+            self.requestId = requestId
+            self.service = serviceConfig.service
+            self.operation = operation
+        }
+    }
+
+    enum RequestKey: BaggageContextKey {
+        typealias Value = RequestMetadata
+        // TODO: the name is not logged as the logger metadata key, check/report/fix
+        var name: String { "aws-sdk" }
+    }
+}
+
+private extension AWSClient.Context {
+    mutating func setRequest(_ value: AWSClient.RequestMetadata) {
+        baggage[AWSClient.RequestKey.self] = value
+    }
+}
+
+// TODO: extensions based on XRaySDK API, temporary, see https://github.com/slashmo/gsoc-swift-tracing/issues/125
+
+private extension TracingInstrument {
```

Comment:
will do, this code will be redundant if the helpers are created per https://github.com/slashmo/gsoc-swift-tracing/issues/125
I dont think its high priority so this may take a while or maybe will never happen

will move to separate file